### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by trace-kadenyi**
- [x] Create a method called current_user in the  `ApplicationController`.
- [x] Create forms to perform the following functions:

1) Creates a Post on behalf of the current_user.
2) Create a comment on behalf of the current_user.
3) Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/posts/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<h1>New Post</h1>
+
+<%= form_with model: @post do |form| %>
```
Comment (SajjadAhmad14) [inline app/views/posts/new.html.erb]: The user should be able to create a post on the behalf of currently logged-in user. Kindly see the error which occurs while trying to create a new post.

post1

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def new
```
Comment (SajjadAhmad14) [inline app/controllers/comments_controller.rb:2]: The user should be able to create a comment on behalf of the currently logged-in user of the app.

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class LikesController < ApplicationController
+  def new
+    @like = Like.new
+  end
+
+  def create
```
Comment (SajjadAhmad14) [inline app/controllers/likes_controller.rb:6]: A user should be able to add a like to a post.

File: config/routes.rb [hunk 4]
```diff
@@ -3,6 +3,9 @@
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
   end
+  resources :posts, only: [:new, :create]
+  resources :comments, only: [:new, :create]
```
Comment (SajjadAhmad14) [inline config/routes.rb]: A comment is added to a specific post so the controller needs a post id. Kindly add new and create routes in such a way that they carry post id with them for comments controller.

File: config/routes.rb [hunk 5]
```diff
@@ -3,6 +3,9 @@
...
+  resources :likes, only: [:new, :create]
```
Comment (SajjadAhmad14) [inline config/routes.rb]: A like is added to a specific post so the controller needs a post id. Kindly add new and create routes in such a way that they carry post id with them for likes controller.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Tracey, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by ShoiraTa** [comment]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @TracK92 , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

### Highlights:
 ✔️ Project is setup correctly
 ✔️ All linters are passing
✔️ Correct gitflow
 ✔️ Professional documentation

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

image

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@shoirata] in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**